### ISSUE TITLE
"RecursionError: maximum recursion depth exceeded" when accessing "upos"

### ISSUE THREAD

**Issue by EmilStenstrom** [EmilStenstrom/conllu#46]
I still observe a strange thing, possibly linked to this workaround: If there are more than two tokens with head 0 in a conllu sentence and I try to get the upos of the sentence root (the top level one), I get a maximum recursion error.
for example the following script reproduces the error:

```
#!/usr/bin/env python3

sent = """1	the	the	DET	_	_	0	root	_	_
2	mouse	mouse	NOUN	_	_	3	root	_	_
3	sleeps	sleep	VERB	_	_	0	root	_	_

"""

for sentence in conllu.parse(sent):
    #print(sentence)
    root = sentence.to_tree()
    root.print_tree()
    print(root.token)
    print(root.token["upos"])
```

Running it gives the following error message
```
(deprel:root) form:_ [0]  # <-- the toplevel root does not have an  UPOS
    (deprel:root) form:the lemma:the upos:DET [1]
    (deprel:root) form:sleeps lemma:sleep upos:VERB [3]
        (deprel:root) form:mouse lemma:mouse upos:NOUN [2]
{'id': 0, 'form': '_', 'deprel': 'root'}
Traceback (most recent call last):
  File "./ex.py", line 15, in <module>
    print(root.token["upos"])
  File "/home/jh/.local/lib/python3.6/site-packages/conllu/models.py", line 31, in __missing__
    return self[self.MAPPING[key]]
  File "/home/jh/.local/lib/python3.6/site-packages/conllu/models.py", line 31, in __missing__
    return self[self.MAPPING[key]]
  File "/home/jh/.local/lib/python3.6/site-packages/conllu/models.py", line 31, in __missing__
    return self[self.MAPPING[key]]
  [Previous line repeated 246 more times]
RecursionError: maximum recursion depth exceeded
```

_Originally posted by @jheinecke in https://github.com/EmilStenstrom/conllu/issues/44#issuecomment-735306647_

**Comment by EmilStenstrom** [comment]
Also: https://github.com/EmilStenstrom/conllu/issues/44#issuecomment-735425268

**Comment by EmilStenstrom** [comment]
@jheinecke Thanks for the research! I'll break this out into a separate issue. That mapping is there to try to be backwards compatible with CoNLL-U version 1. I like to support both versions simultaneously, so I need to find a way to just do one mapping and stop.

**Comment by jheinecke** [comment]
You could use internally always "upos" (for CoNLL-U v1 and v2 and make only the API accept both ?

**Comment by EmilStenstrom** [comment]
Thanks for reporting this @jheinecke. This is now fixed in version 4.2.2.
